### PR TITLE
Update jsonschema and define "services"

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

File: pipeline/schema/schema.json [hunk 1]
```diff
@@ -304,13 +258,118 @@
...
+          "$ref": "#/definitions/step_settings"
+        },
+        "when": {
+          "$ref": "#/definitions/step_when"
+        },
+        "volumes": {
+          "$ref": "#/definitions/step_volumes"
+        },
+        "ports": {
+          "description": "export port steps in pipeline can connect to",
```
Comment (6543) [inline pipeline/schema/schema.json:366]:
```suggestion
          "description": "export ports, witch steps can connect to",
```
?

Comment (6543) [reply pipeline/schema/schema.json:366]: -> I have to add it to the schema or our pipeline linting step will fail - wo do use it :sweat_smile:

File: pipeline/schema/schema.json [hunk 2]
```diff
@@ -304,13 +258,118 @@
...
+      "description": "Change the settings of your plugin. Read more: https://woodpecker-ci.org/docs/usage/plugins/plugins",
+      "type": "object",
+      "additionalProperties": {
+        "type": ["boolean", "string", "number", "array", "object"]
+      }
+    },
+    "step_volumes": {
+      "description": "Mount files or folders from the host machine into your step container. Read more: https://woodpecker-ci.org/docs/usage/volumes",
+      "oneOf": [
+        { "type": "string" },
```
[the remark this answers is not in the document]
Comment (6543) [reply pipeline/schema/schema.json:324]: ? either string or array of strings

File: pipeline/schema/schema.json [hunk 3]
```diff
@@ -304,13 +258,118 @@
...
     "services": {
-      "description": "TODO Read more: https://woodpecker-ci.org/docs/usage/services",
+      "description": "Read more: https://woodpecker-ci.org/docs/usage/services",
       "type": "object",
-      "additionalProperties": true
+      "additionalProperties": { "$ref": "#/definitions/service" },
+      "minProperties": 1
+    },
+    "service": {
+      "description": "Read more: https://woodpecker-ci.org/docs/usage/pipeline-syntax#steps",
```
[the remark this answers is not in the document]
Comment (6543) [reply pipeline/schema/schema.json:335]: https://woodpecker-ci.org/docs/usage/services ?

File: pipeline/schema/schema.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (anbraten) [reply pipeline/schema/schema.json:324]: Array make sense, but why is a single string allowed for this again 🤦🏾

```yml
volumes: /etc/ssl/certs:/etc/ssl/certs
```

As this version with just the string isn't documented (at least as far as I can tell), I would directly drop that support. What do you think?

[the remark this answers is not in the document]
Comment (6543) [reply pipeline/schema/schema.json:324]: Unmarshall will support it but we can remove it from shema to make clear this is a bad habit ...

[the remark this answers is not in the document]
Comment (anbraten) [reply pipeline/schema/schema.json:324]: I think that would be good

**Review by anbraten — APPROVED** [review]

**Pull request merged**